src/dist/replication/lib/replica_stub.cpp

```diff
@@ -1231,7 +1257,36 @@ void replica_stub::close()
         _log->close();
         _log = nullptr;
     }
+}
 
+std::string replica_stub::get_replica_dir(const char* app_type, global_partition_id gpid) const
```

Comment:
Yes, I have also considered the problem, but it is not easy to think out a perfect assignment strategy. The mapping should be deterministic and extensible, considering adding or removing some data directories when restating the app. So now I just check all data directories when new/load replica, referring to replica_stub::get_replica_dir(). We can improve it if think out a better strategy.